pkg/handler/eval_test.go

```diff
@@ -38,6 +40,59 @@ func TestEvalSegment(t *testing.T) {
 		assert.False(t, evalNextSegment)
 	})
 
+	t.Run("evalSegment attaches EvalContext members to EntityContext with '@' prefix", func(t *testing.T) {
+		entityID := "entityID1"
+		s := entity.GenFixtureSegment()
+		s.Constraints = append(s.Constraints, entity.Constraint{
+			Model:     gorm.Model{ID: 501},
+			SegmentID: 200,
+			Property:  "@entityID",
+			Operator:  models.ConstraintOperatorEQ,
+			Value:     fmt.Sprintf(`"%s"`, entityID),
+		})
+		s.RolloutPercent = uint(100)
+		s.PrepareEvaluation()
+
+		vID, log, evalNextSegment := evalSegment(100, models.EvalContext{
+			EnableDebug:   true,
+			EntityContext: map[string]interface{}{"dl_state": "CA"},
+			EntityID:      entityID,
+			EntityType:    "entityType1",
+			FlagID:        int64(100),
+		}, s)
+
+		assert.NotNil(t, vID)
+		assert.NotEmpty(t, log)
+		assert.False(t, evalNextSegment)
+	})
+
+	t.Run("evalSegment attaches Config.EvalServerEntityContext to EntityContext", func(t *testing.T) {
+		serverEntityContext := map[string]string{"foo": "bar"}
+		config.Config.EvalServerEntityContext = serverEntityContext
```

Comment:
please reset the config (ideally with a defer) once the test is done. this will minimize the side effect of future tests

Reply:
good call, done